charts/patch/templates/patch.json

```diff
@@ -44,14 +44,14 @@
   },
   {{- end }}
   {{- end}}
-  {{- if and .Values.proxyInit (not .Values.cniEnabled) }}
```

Comment:
To put this another way, we'll always add an init container: either the noop if cni is enabled or the proxy-init if cni is not enabled.

Technically, we could skip adding the noop container if the spec already contains some other init container, but I think I'd prefer to keep it simple rather than conditioning on that.